python/mlc_chat/support/tensor_parallel.py

```diff
@@ -8,208 +7,91 @@
 
 
 @dataclasses.dataclass
-class Row:
-    """Shard a 2D tensor by its rows."""
+class Shard1Dim:
```

Comment:
do we have anything other than `Shard1D`? I suppose the `1Dim` here means "a single dimenison"? If so, how about `ShardSingleDim`?

Reply:
`ShardSingleDim` sounds good for me